winner-tool/src/main.rs

```diff
@@ -0,0 +1,179 @@
+        )
+        .arg(
+            Arg::with_name("bootstrap_leader")
+                .long("bootstrap-leader")
+                .value_name("PUBKEY")
+                .takes_value(true)
+                .required(true)
+                .validator(is_pubkey)
+                .help("Public key of the bootstrap leader"),
+        )
+        .arg(
+            Arg::with_name("final_slot")
+                .long("final-slot")
+                .value_name("SLOT")
+                .takes_value(true)
+                .help("Final slot of TdS ledger"),
+        )
+        .get_matches();
+
+    let ledger_path = PathBuf::from(value_t_or_exit!(matches, "ledger", String));
+    let starting_balance_sol = value_t_or_exit!(matches, "starting_balance", f64);
+    let baseline_id_string = value_t_or_exit!(matches, "baseline_validator", String);
+    let bootstrap_id_string = value_t_or_exit!(matches, "bootstrap_leader", String);
+    let final_slot = value_t!(matches, "final_slot", u64).ok();
+
+    let baseline_id = Pubkey::from_str(&baseline_id_string).unwrap_or_else(|err| {
+        eprintln!(
+            "Failed to create a valid pubkey from {}: {}",
+            baseline_id_string, err
+        );
+        exit(1);
+    });
+
+    let bootstrap_id = Pubkey::from_str(&bootstrap_id_string).unwrap_or_else(|err| {
+        eprintln!(
+            "Failed to create a valid pubkey from {}: {}",
+            bootstrap_id_string, err
+        );
+        exit(1);
+    });
+
+    let genesis_block = GenesisBlock::load(&ledger_path).unwrap_or_else(|err| {
+        eprintln!(
+            "Failed to open ledger genesis_block at {:?}: {}",
+            ledger_path, err
+        );
+        exit(1);
+    });
+
+    let blocktree = match Blocktree::open(&ledger_path) {
+        Ok(blocktree) => blocktree,
+        Err(err) => {
+            eprintln!("Failed to open ledger at {:?}: {}", ledger_path, err);
+            exit(1);
+        }
+    };
+
+    // Track voter record after each entry
+    let voter_record: Arc<RwLock<VoterRecord>> = Arc::default();
+    let slot_voter_segments: Arc<RwLock<SlotVoterSegments>> = Arc::default();
+    let entry_callback = {
+        let voter_record = voter_record.clone();
+        let slot_voter_segments = slot_voter_segments.clone();
+        Arc::new(move |bank: &Bank| {
+            confirmation_latency::on_entry(
+                bank.slot(),
+                bank.vote_accounts(),
+                &mut voter_record.write().unwrap(),
+                &mut slot_voter_segments.write().unwrap(),
+            );
+        })
+    };
+
+    let opts = ProcessOptions {
+        verify_ledger: false,
+        dev_halt_at_slot: final_slot,
+        full_leader_cache: true,
+        entry_callback: Some(entry_callback),
+        override_num_threads: Some(1),
+    };
+
+    let excluded_set = {
+        let mut set = HashSet::new();
+        set.insert(baseline_id);
+        set.insert(bootstrap_id);
```

Comment:
I think we should rename `--bootstrap-leader` to `--exclude XXX`, and support multiple excludes.  Solana runs a several nodes currently in the TdS cluster, we'd want to exclude them all

Reply:
https://github.com/solana-labs/tour-de-sol/issues/14

Reply:
Done